controller/lib/src/external_api/workload/controller.rs

```diff
@@ -1,46 +1,318 @@
-    service::create_workload(&workload_info);
-    HttpResponse::Ok().body("put_workload")
-}
+impl WorkloadController {
+    pub fn get_services(&self) -> Scope {
+        web::scope("/workload")
+            .service(
+                web::resource("/{namespace}/{workload_id}")
+                    .route(web::delete().to(WorkloadController::delete_workload))
+                    .route(web::get().to(WorkloadController::get_workload))
+                    .route(web::patch().to(WorkloadController::patch_workload)),
+            )
+            .service(
+                web::resource("/{namespace}")
+                    .route(web::put().to(WorkloadController::put_workload))
+                    .route(web::get().to(WorkloadController::get_all_workloads)),
+            )
+    }
+
+    /// `get_workload` is an async function that handle **/workload/\<namespace>/<workload_id>** route (GET)
+    /// # Description:
+    /// * Get a workload by id and namespace
+    /// # Arguments:
+    ///
+    /// * `params`: web::Path<(String, String)> - The first Path parameter is the namespace and the second the workload id.
+    pub async fn get_workload(
+        params: web::Path<(String, String)>,
+        data: web::Data<ActixAppState>,
+    ) -> impl Responder {
+        let (namespace, workload_id) = params.into_inner();
+
+        let mut workload_service = service::WorkloadService::new(&data.etcd_address)
+            .await
+            .map_err(|err| {
+                let mut http_response = HttpResponse::InternalServerError();
+                match err {
+                    WorkloadError::Etcd(msg) => http_response.body(format!("Etcd error: {} ", msg)),
+                    _ => http_response.body("Internal Server Error"),
+                }
+            })
+            .unwrap();
+
+        match workload_service
+            .get_workload(&workload_id, &namespace)
+            .await
+        {
+            Ok(workload) => match serde_json::to_string(&workload) {
+                Ok(json) => HttpResponse::build(StatusCode::OK).body(json),
+                Err(err) => HttpResponse::InternalServerError().body(format!(
+                    "Error while converting the workload to json string : {}",
+                    err
+                )),
+            },
+            Err(e) => match e {
+                WorkloadError::WorkloadNotFound => {
+                    HttpResponse::build(StatusCode::NOT_FOUND).body("Workload not found")
+                }
+                WorkloadError::JsonToWorkload(msg) => {
+                    HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).body(format!(
+                        "Error while converting JSON string to workload : {}",
+                        msg
+                    ))
+                }
+                _ => HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR)
+                    .body("Internal Server Error"),
+            },
+        }
+    }
```

Comment:
Similar comment here: Please define and use a `WorkloadError`converter method. Since you don't need any specific parameter, you should use the `From`trait directly: https://doc.rust-lang.org/std/convert/trait.From.html

Reply:
Yes indeed, I implementedthe function `to_http` for WorkloadError enum. 
It returns the corresponding http error, What do you think about it ?